pkgs/development/tools/godot/4/default.nix

```diff
@@ -86,6 +87,33 @@ stdenv.mkDerivation rec {
 
   enableParallelBuilding = true;
 
+  # Usually the version status is set to "stable". However, for betas and rcs,
+  # this is set to "beta"/"rc" plus the number of the beta/rc release.
+  #
+  # Once Godot 4 is released, this no longer has to be specified manually.
+  #
+  # See also 'methods.py' in the Godot repo and 'status' in
+  # https://docs.godotengine.org/en/latest/classes/class_engine.html#class-engine-method-get-version-info
+  GODOT_VERSION_STATUS = "rc1";
+
+  # Set the build name which is part of the version. In official downloads, this
+  # is set to 'official'. On gentoo, it is set to 'gentoo'. When not specified
+  # explicitly, it is set to 'custom_build'.
+  #
+  # See also 'methods.py' in the Godot repo and 'build' in
+  # https://docs.godotengine.org/en/latest/classes/class_engine.html#class-engine-method-get-version-info
+  BUILD_NAME = "nixos";
```

Comment:
I’m now realizing that `BUILD_NAME` should probably changed to `nixpkgs`. I haven’t tried it, but I’m assuming that you can use `godot_4` on macOS or a Linux distro other than NixOS.

Reply:
In addition to what you said, NixOS is technically a part of nixpkgs, not the other way around. You convinced me.